internal/app/storage/mapstorage.go

```diff
@@ -0,0 +1,34 @@
+package storage
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// MapStorage defines a simple in-memory storage implemented as a wrapper aroung Go map
+// FIXME not thread-safe at the moment
+type MapStorage struct {
+	m map[string]string
+}
+
+// MapStorage implements Storage interface
+var _ Storage = (*MapStorage)(nil)
+
+func NewMapStorage() *MapStorage {
+	m := make(map[string]string)
+	return &MapStorage{m}
+}
+
+func (st *MapStorage) AddURL(url string) (id string) {
+	// TODO There should be more sophisticated algorithm for calculating URL's short id
```

Comment:
You can use a "standard" approach: UUID (the collision factor is good enough for this task).